modules/order/src/Form/MultipleProfileTypesConfirmForm.php

```diff
@@ -0,0 +1,442 @@
+        running versions that are incompatible with using multiple profile types
+        and it could possibly render the site as unusable.
+        <p><strong>@modules</strong></p>
+        <p>Please upgrade and try again.</p>', [
+          '@modules' => implode('<br>', $incompatible_modules)
+        ]
+      ));
+
+      return;
+    }
+
+    $this->messenger()->addWarning($this->t('
+      If you choose to proceed, profiles for existing orders will be migrated to
+      use separate profile types. That can take some time and it might cause 
+      errors if the users try to view their existing orders during the process;
+      if you are running this on a production site please make sure that you are
+      in maintenance mode.'
+    ));
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    /** @var \Drupal\commerce_order\Entity\OrderTypeInterface $order_type */
+    $order_type = $this->entity;
+
+    // Create the billing and shipping profile types.
+    $this->createProfileTypes();
+
+    // Batch process to migrate the existing order profiles from the customer
+    // profile type to use the billing/shipping profile types.
+    $this->migrateExistingProfiles();
+
+    // Set the useMultipleProfileTypes field to TRUE now that we've processed
+    // everything.
+    $order_type->setUseMultipleProfileTypes(TRUE);
+    $order_type->save();
+
+    $form_state->setRedirectUrl($order_type->toUrl('collection'));
+  }
+
+  /**
+   * Create the billing and shipping profile types, if not already created.
+   */
+  public function createProfileTypes() {
+    $profile_types = [
+      OrderType::PROFILE_BILLING => $this->t('Customer Billing'),
+      OrderType::PROFILE_SHIPPING => $this->t('Customer Shipping'),
+    ];
+
+    // Load the 'customer' profile type so we can just duplicate that for the
+    // new billing and shipping profile types.
+    $profile_type_storage = $this->entityTypeManager->getStorage('profile_type');
+    /** @var \Drupal\commerce_order\Entity\OrderType $order_type */
+    $customer_profile_type = $profile_type_storage->load(OrderType::PROFILE_COMMON);
+
+    // Fetch the non-base fields from the 'customer' profile type so we can copy
+    // the same to the new profile types.
+    $extra_fields_to_add = [];
+    $field_definitions = $this->entityFieldManager->getFieldDefinitions('profile', OrderType::PROFILE_COMMON);
+    foreach ($field_definitions as $field_name => $field_definition) {
+      if ($field_definition->getFieldStorageDefinition()->isBaseField()) {
+        continue;
+      }
+
+      $extra_fields_to_add[$field_name] = $field_definition;
+    }
+
+    // Now create the profile type.
+    foreach ($profile_types as $id => $profile_type) {
+      $this->createProfileType([$id => $profile_type], $customer_profile_type, $extra_fields_to_add);
```

Comment:
@krystalcode This point onwards there's some changes to the original code. We're not using config files to import the billing and shipping profile types. Instead, we're basically duplicating the 'customer' profile types as is. We're also copying the entity_view_display and entity_form_display modes as I noticed that that was not done. This ensures that shipping and billing displays exactly the same as before and the admin won't see a difference.